src/main/java/org/bytecodeparser/instruction/PrettyPrintTypes.java

```diff
@@ -0,0 +1,42 @@
+package org.bytecodeparser.instruction;
+
+import org.bytecodeparser.utility.ThrowingFunction;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.function.Function;
+
+public enum PrettyPrintTypes {
+    SIMPLE(dataInputStream -> InstructionArguments.builder().build(), arguments -> ""),
```

Comment:
for every SIMPLE instruction you use InstructionArguments.builder().build(). Every time allocating object but there is no need in this - factor out it in the static singleton instance and reuse everytime

Reply:
this was done to unify all Function<DataInputStream, InstructionArguments> resolveStrategy with each other
unify prefer efficiency in this case. IMHO.

But I try to implement your idea and it looks bad. Check it.

Reply:
move it to InstructionArguments class

Reply:
will do it with follow up commit by myself